src/librustc_codegen_ironox/back/write.rs

```diff
@@ -8,14 +8,55 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-#![allow(dead_code)]
+use errors::{Handler, FatalError};
+use rustc::session::config::OutputType;
+use rustc::util::time_graph::Timeline;
+use rustc_codegen_ssa::{ModuleCodegen, CompiledModule};
+use rustc_codegen_ssa::back::write::{CodegenContext, ModuleConfig, run_assembler};
 
-macro_rules! asm {
-    ($m:expr, $($args:expr)*) => {
-        $(
-            $m.asm.push_str(&format!("{}\n", $args));
-        )*
+use std::fs;
+use std::path::PathBuf;
+
+use IronOxCodegenBackend;
+use ModuleIronOx;
+
+/// Write the assembler instructions from a `module` to the specified `path`.
+fn write_module(
+    module: &ModuleIronOx,
+    path: &PathBuf,
+    diag_handler: &Handler) {
+    if let Err(e) = fs::write(path, module.asm()) {
+        diag_handler.err(&format!("failed to write asm file {}", e));
     }
 }
 
-// FIXME not yet implemented
+pub unsafe fn codegen(
+    cgcx: &CodegenContext<IronOxCodegenBackend>,
+    diag_handler: &Handler,
+    module: ModuleCodegen<ModuleIronOx>,
+    config: &ModuleConfig,
+    timeline: &mut Timeline
+) -> Result<CompiledModule, FatalError> {
+    // no_integrated_as has to be true in order for the IronOx backend to work
```

Comment:
What is an integrated assembler in this context?

Check your comments for full stops.

Reply:
It is the assembler that comes with LLVM. The integrated assembler allows rustc to emit object files directly. Without the integrated assembler, it would have to first output the assembly instructions to a file, and then run the assembler. 

Setting `no_integrated_as` to `true` means `gcc` will be used to create the object files (from the '.s' files produced by `wite_module`).

Reply:
OK.

I wonder if it would be better to panic at the top of the function if there's no integrated asm, then the rest of the function can be de-dented once for better readability. What do you think?

```
if config.no_integrated_as {
    panic!("no integrated assembler!");
}

...rest of function
```

Reply:
I agree, that would probably be a bit more readable.